src/daemon/taproxy.rs

```diff
@@ -736,6 +767,23 @@ impl TrustAnchorProxy {
         }
     }
 
+    fn process_update_signer(
+        &self,
+        signer: TrustAnchorSignerInfo,
+    ) -> KrillResult<Vec<TrustAnchorProxyEvent>> {
+        if let Some(s) = &self.signer {
+            if s.ta_cert_details.cert().key_identifier() == 
+                signer.ta_cert_details.cert().key_identifier() {
+                // It is not possible to add a signer that has a different
+                // public key
+                return Ok(vec![
+                    TrustAnchorProxyEvent::SignerAdded(signer)
```

Comment:
Should this have its own event rather than re-using Signer Added? (Honest question, I’m not sure.)